src/chi-vue/src/components/data-table/DataTable.tsx

```diff
@@ -0,0 +1,852 @@
+    } else {
+      this.slicedData.forEach((row: DataTableRow) => {
+        if (row.id) {
+          const rowIndex = this.selectedRows.indexOf(`row-${this._dataTableNumber}-${row.id}`);
+
+          this.selectedRows.splice(rowIndex, 1);
+        }
+      });
+    }
+    this.$emit(DATA_TABLE_EVENTS.SELECTED_ROWS_CHANGE, this.selectedRows);
+  }
+
+  selectRowCheckbox(selectAll: boolean, rowId = '') {
+    const checkedState = !!rowId && this.selectedRows.includes(rowId);
+
+    return (
+      <div
+        class={`
+        ${DATA_TABLE_CLASSES.CELL}
+        ${DATA_TABLE_CLASSES.SELECTABLE}
+      `}
+      >
+        <div class={CHECKBOX_CLASSES.checkbox}>
+          <input
+            type="checkbox"
+            class={CHECKBOX_CLASSES.INPUT}
+            id={`row-${this._dataTableNumber}-${rowId}`}
+            onChange={(ev: Event) => {
+              if (selectAll) {
+                this.selectAllRows(
+                  (ev.target as HTMLInputElement).checked
+                    ? 'select'
+                    : 'deselect'
+                );
+              } else {
+                if (rowId) {
+                  this.selectRow(rowId);
+                }
+              }
+            }}
+            checked={checkedState}
+          />
+          <label class={CHECKBOX_CLASSES.LABEL} for={`row-${this._dataTableNumber}-${rowId}`}>
+            <div class={SR_ONLY}>Select row {rowId}</div>
+          </label>
+        </div>
+      </div>
+    );
+  }
+
+  toggleRow(id: string) {
+    if (this.accordionsShown.includes(id)) {
+      this.accordionsShown.splice(this.accordionsShown.indexOf(id), 1);
+    } else {
+      this.accordionsShown.push(id);
+    }
+  }
+
+  rowAccordionContent(accordionData: any) {
+    if (accordionData.template) {
+      const template = this.$scopedSlots[accordionData.template]!(
+        accordionData.payload
+      );
+
+      return (
+        <div class={`${DATA_TABLE_CLASSES.ROW_CHILD} -p--2`} role="row">
+          {template}
+        </div>
+      );
+    } else if (accordionData.table) {
+      if (accordionData.table.data) {
+        return accordionData.table.data.map((bodyRow: DataTableRow) => {
+          return this.row(bodyRow, true);
+        });
+      }
+    } else {
+      return (
+        <div class={`${DATA_TABLE_CLASSES.ROW_CHILD} -p--2`} role="row">
+          {accordionData.value}
+        </div>
+      );
+    }
+  }
+
+  row(bodyRow: DataTableRow, rowChild = false, striped = false) {
+    const row = [],
+      rowCells = [],
+      rowAccordionContent = [],
+      rowId = bodyRow.id ? `row-${this._dataTableNumber}-${bodyRow.id}` : `row-${this._dataTableNumber}-${uuid4()}`,
+      rowClass = rowChild
+        ? DATA_TABLE_CLASSES.ROW_CHILD
+        : DATA_TABLE_CLASSES.ROW;
+
+    if (this._expandable) {
+      if (bodyRow.nestedContent) {
+        rowCells.push(this.expansionButton(rowId));
+        rowAccordionContent.push(this.rowAccordionContent(bodyRow.nestedContent));
```

Comment:
Does this row accordion content also supports the nested rows?.

Reply:
Yes, nested row is supported. Functionally, on infinite levels (the PR example shows 5 level nesting, though, though UX-UI team requires Table to support maximum 3 level nesting and we are still collaborating on the visual styling on each level).